src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,56 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<F, S> {
+    private F first;
+    private S second;
 
+    private Pair(F first, S second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public static <F, S> Pair of(F first, S second) {
+        return new Pair(first, second);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || !(o.getClass().equals(Pair.class))) {
+            return false;
+        }
+        Pair<?, ?> pair = (Pair<?, ?>) o;
```

Comment:
what's the purpose of using wildcard here?

Reply:
It was generated by IDEA, but I've changed this. In the future, it's better to do like now, yes?

Reply:
when we know the types we work with in the method, we should use them
in these rare cases we may ignore IDEA warnings

this is our method implementation, and we can ensure ourselves that this cast is secure